spock-core/src/main/java/spock/util/time/MutableClock.java

```diff
@@ -0,0 +1,167 @@
+package spock.util.time;
+
+import java.time.*;
+import java.time.temporal.*;
+import java.util.function.BiFunction;
+
+/**
+ * A mutable implementation of {@link Clock} for testing purposes.
+ *
+ * @author Leonard Brünings
+ * @since 2.0
+ */
+public class MutableClock extends Clock {
+
+  private final ZoneId zone;
+  private Instant instant;
+
+  /**
+   * {@link TemporalAmount} used for {@link #next()} and {@link #previous()} changes.
+   */
+  private TemporalAmount changeAmount;
+
+  /**
+   * Initializes a with {@link Instant#now()} and {@link ZoneId#systemDefault()} and 1 second {@link #changeAmount}.
+   */
+  public MutableClock() {
+    this(Instant.now(), ZoneId.systemDefault());
+  }
+
+  /**
+   * Initializes a with given instant and {@link ZoneId#systemDefault()} and 1 second {@link #changeAmount}.
+   * @param instant to use
+   */
+  public MutableClock(Instant instant) {
+    this(instant, ZoneId.systemDefault());
+  }
+
+  /**
+   * Initializes a with {@link Instant#now()} and given zone and 1 second {@link #changeAmount}.
+   * @param zone to use
+   */
+  public MutableClock(ZoneId zone) {
+    this(Instant.now(), zone);
+  }
+
+  /**
+   * Initializes a with 1 second {@link #changeAmount} and given parameters.
+   * @param instant to use
+   * @param zone to use
+   */
+  public MutableClock(Instant instant, ZoneId zone) {
+    this(instant, zone, Duration.ofSeconds(1));
+  }
+
+  public MutableClock(Instant instant, ZoneId zone, TemporalAmount changeAmount) {
+    assert instant != null : "instant may not be null";
+    assert zone != null : "zone may not be null";
+    assert changeAmount != null : "changeAmount may not be null";
```

Comment:
It's Java, it will not work by default.

`Objects.requireNonNull(instant, "instant may not be null")`?

Reply:
Well this is only intended to run in tests, and you usually start those with `-ea` so it would work, but I changed it.

Reply:
I forgot about the context, but thanks anyway.

Unfortunately, I've just spotted that there are also `assert`s in two setters in the same class (that I missed previously). What do you think about changing also them, for consistency?

Reply:
done